assignment2/src/app/core/auth/auth.service.ts

```diff
@@ -0,0 +1,40 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+import { catchError, tap } from 'rxjs/operators';
+import { BehaviorSubject, throwError } from 'rxjs';
+import { UserModel } from '../../shared/user.model';
+
+export interface LoginData {
+  email: string;
+  password: string;
+}
```

Comment:
Use class in separate file instead of interface.

Reply:
e chưa hiểu cách fix??

Reply:
Dùng class thay vì interface và tách nó ra thành một file.